smlb/decomposition/pca_prince.py

```diff
@@ -0,0 +1,105 @@
+from copy import deepcopy
+from typing import Optional
+
+import prince
+
+from smlb import (
+    DataValuedTransformation,
+    InvertibleTransformation,
+    Data,
+    TabularData,
+    params,
+)
+
+
+class PCAPrince(DataValuedTransformation, InvertibleTransformation):
+    """Principal component analysis (PCA), prince implementation.
+
+    .. seealso::
+        See `here <https://github.com/MaxHalford/prince#principal-component-analysis-pca>`_ for full documentation.
+    """
+
+    def __init__(
+        self,
+        rng: Optional[int] = None,
+        rescale_with_mean: bool = True,
+        rescale_with_std: bool = True,
+        n_components: int = 2,
+        n_iter: int = 3,
+        inverse_transform: bool = False,
+        *args,
+        **kwargs
+    ):
+        """Initialize state.
+
+        Parameters:
+            rng: Random state
+            rescale_with_mean: Whether to subtract each column's mean or not.
+            rescale_with_std: Whether to divide each column by it's standard deviation or not.
+            n_components: The number of principal components to compute.
+            n_iter: The number of iterations used for computing the SVD.
```

Comment:
```suggestion
            n_iter: The number of power iterations used for computing the SVD.
```

So few iterations are usually not sufficient for typical applications of the power method, so I was curious whether this might be enough. Tl;dr: probably. See page 10 of https://arxiv.org/pdf/0909.4061.pdf